### PR TITLE
Fix is_shib_auth variable to use third party auth

### PR DESCRIPTION
**Pull request by kluo**
@stvstnfrd @caesar2164 

Should this be for any saml provider not just sunet?

### CODE REVIEW

File: lms/djangoapps/student_account/views.py [hunk 1]
```diff
@@ -504,10 +501,9 @@ def account_settings_context(request):
             # in with, or if the user is already authenticated with them.
         } for state in auth_states if state.provider.display_for_login or state.has_account]
 
-    try:
-        external_auth_map = ExternalAuthMap.objects.get(user=user)
-    except:
-        external_auth_map = None
-    context['is_shib_auth'] = 'shib' in external_auth_map.external_domain if external_auth_map else False
```
Comment (stvstnfrd) [inline lms/djangoapps/student_account/views.py]: @kluo Ah, so this is an issue of switching from `external_auth` to `third_party_auth`?

**Comment by kluo** [comment]
test this please

File: lms/djangoapps/student_account/views.py (hunk 1 above)
Comment (kluo) [reply lms/djangoapps/student_account/views.py]: Correct

**Comment by stvstnfrd** [comment]
> Should this be for any saml provider not just sunet?

For backwards compatibility,
we should limit it to sunet.

File: lms/templates/student_account/account_settings.html [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
 <%block name="js_extra">
 <%static:require_module module_name="js/student_account/views/account_settings_factory" class_name="AccountSettingsFactory">
-    var isShibAuth = ${ json.dumps(is_shib_auth) };
+    var isShibAuth = ${ is_shib_auth | n, dump_js_escaped_json };
```
Comment (stvstnfrd) [inline lms/templates/student_account/account_settings.html]: Why this change?

Comment (kluo) [reply lms/templates/student_account/account_settings.html]: Keeping up-to-date with upstream's changes to the other subsequent vars

**Comment by kluo** [comment]
test this please

**Comment by stvstnfrd** [comment]
+1
